tests/js/session-tests.js

```diff
@@ -1041,6 +1051,114 @@ module.exports = {
                             resolve('Done');
                         }
                     });
+                    setTimeout(() => { reject() }, 10000);
+                }).catch(error => reject(error));
+            });
+        });
+    },
+
+    testStartStop() {
+        if(!isNodeProccess) {
+            return;
+        }
+
+        return Realm.Sync.User.register('http://localhost:9080', uuid(), 'password').then((u) => {
+            return new Promise((resolve, reject) => {
+                const config = {
+                    sync: {
+                        user: u,
+                        url: `realm://localhost:9080/~/${uuid()}`,
+                        fullSynchronization: true,
+                    }
+                };
+
+                const checks = {
+                    started: false,
+                    stopped: false,
+                    restarted: false,
+                }
+
+                Realm.open(config).then(realm => {
```

Comment:
You can move the `Realm.open` outside the promise constructor. Something like:

```js
Realm.Sync.User.register(...)
    .then((user) => {
        const config = { ... };
        return Realm.open(config);
    })
    .then((realm) => {
        return new Promise(...);
    });
```

Reply:
I see all the other tests in the file are structured as mine. Also, I cannot concatenate a `.catch(error => reject(error));` at the end of the chain you propose.

Reply:
You don't need to concatenate it because the promise will be rejected anyway.

Reply:
With the current code, you reject a promise if `Realm.open` fails. If you return it as I suggest, you'll end up with the same logical flow without having a codepath for promise rejection.